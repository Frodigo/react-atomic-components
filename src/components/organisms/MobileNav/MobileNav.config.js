export const items = [
    {
        label: 'Home',
        image: {
            alt: 'Home',
            url: 'https://via.placeholder.com/20x20',
        },
        url: '/',
        id: 1
    },
    {
        label: 'Blog',
        image: {
            alt: 'Blog',
            url: 'https://via.placeholder.com/20x20',
        },
        url: '/blog',
        id: 2
    },
    {
        label: 'About',
        image: {
            alt: 'About',
            url: 'https://via.placeholder.com/20x20',
        },
        url: '/about-me',
        id: 7
    },
    {
        label: 'Contact',
        image: {
            alt: 'Contact',
            url: 'https://via.placeholder.com/20x20',
        },
        url: '/get-in-touch',
        id: 8
    },
    {
        label: 'Subscribe',
        image: {
            alt: 'Subscribe',
            url: 'https://via.placeholder.com/20x20',
        },
        url: '#newsletter',
        id: 9
    }
];
