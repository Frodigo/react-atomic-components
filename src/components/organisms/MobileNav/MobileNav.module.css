.mobileNav {
    align-items: center;
    background-color: var(--color-text);
    bottom: 0;
    display: flex;
    justify-content: space-between;
    left: 0;
    padding: 0 var(--spacing-xs);
    position: fixed;
    right: 0;
    z-index: var(--z-index-largest);
}

.submenu {
    align-items: center;;
    background-color: var(--color-primary);
    bottom: -100%;
    display: flex;
    justify-content: flex-end;
    flex-wrap: wrap;
    flex-direction: column;
    left: 0;
    padding: var(--spacing-xl);
    position: fixed;
    right: 0;
    transition: .3s bottom;
}

.submenuOpened {
    composes: submenu;
    bottom: 50px;
}

.submenuItems {
    display: flex;
    flex-wrap: wrap;
    width: 100%;
}

.submenuItem {
    flex: 1 1 50%;
    text-decoration: none;
}


.submenuItem > span {
    color: var(--color-white);
    font-size: 14px;
    font-weight: var(--font-weight-bold);
    justify-content: center;
    height: 100px;
}

.closeButton,
.submenuTrigger {
    background: transparent;
    border: 0;
    cursor: pointer;
    outline: none;
}


.submenuHeader {
    align-items: center;
    border-bottom: 2px solid var(--color-white);
    display: flex;
    justify-content: center;
    padding-bottom: 15px;
    width: 100%;
}

.submenuHeader h2 {
    font-weight: 700;
}

@media (min-width: 1024px) {
    .mobileNav {
        display: none;
    }
}