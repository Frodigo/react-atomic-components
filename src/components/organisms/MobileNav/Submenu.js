import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Link from '../../atoms/Link';
import IconWithText from '../../molecues/IconWithText';
import Heading from '../../atoms/Text/Heading';

import classes from './MobileNav.module.css';

const Submenu = (props) => {
    const { item } = props;
    const [isSubmenuOpen, setIsSubmenuOpen] = useState(false);

    const handleSubmenu = () => {
        setIsSubmenuOpen(!isSubmenuOpen);
    };

    const submenuItems = item.submenu.map((submenuItem) => (
        <a
            href={submenuItem.url}
            key={submenuItem.id}
            className={classes.submenuItem}
        >
            <IconWithText label={submenuItem.label} image={submenuItem.image} />
        </a>
    ));

    return (
        <>
            <button
                type="button"
                onClick={handleSubmenu}
                className={classes.submenuTrigger}
            >
                <IconWithText
                    label={item.label}
                    image={item.image}
                    isActive={isSubmenuOpen}
                />
            </button>

            <nav
                className={
                    isSubmenuOpen ? classes.submenuOpened : classes.submenu
                }
            >
                <header className={classes.submenuHeader}>
                    <Heading classes="inverse" type="h2">
                        Read about:
                    </Heading>
                </header>

                <div className={classes.submenuItems}>{submenuItems}</div>

                <Link
                    variant="tiertary"
                    href={item.url}
                    style={{ marginTop: '30px' }}
                >
                    Show all
                </Link>
            </nav>
        </>
    );
};

Submenu.propTypes = {
    item: PropTypes.object.isRequired
};

export default Submenu;
