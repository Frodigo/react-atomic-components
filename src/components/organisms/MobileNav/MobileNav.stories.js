import React from 'react';

import { default as MobileNavComponent } from './MobileNav';
import { items } from './MobileNav.config';

export default {
    title: 'Organisms/MobileNav',
    component: MobileNavComponent
};

const Template = (args) => <MobileNavComponent {...args} />;

export const Default = Template.bind({});
Default.args = {
    items
};
