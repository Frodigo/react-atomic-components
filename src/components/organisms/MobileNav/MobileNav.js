import React from 'react';
import PropTypes from 'prop-types';

import IconWithText from '../../molecues/IconWithText';

import Submenu from './Submenu';
import classes from './MobileNav.module.css';

const MobileNav = (props) => {
    const { items, ...rest } = props;
    const navItems = items.map((item) => {
        if (!item.submenu) {
            return (
                <a href={item.url} key={item.label}>
                    <IconWithText label={item.label} image={item.image} />
                </a>
            );
        }

        return <Submenu item={item} key={item.label} />;
    });

    return (
        <nav className={classes.mobileNav} {...rest}>
            {navItems}
        </nav>
    );
};

MobileNav.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            image: PropTypes.shape({
                alt: PropTypes.string,
                url: PropTypes.string
            }).isRequired,
            id: PropTypes.number.isRequired,
            label: PropTypes.string.isRequired,
            url: PropTypes.string.isRequired
        })
    ).isRequired
};

export default MobileNav;
