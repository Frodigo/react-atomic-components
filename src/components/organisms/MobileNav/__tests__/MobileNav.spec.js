import React from 'react';
import { render } from '@testing-library/react';

import { items } from '../MobileNav.config';
import MobileNav from '../MobileNav';

describe('Mobile nav should be rendered', () => {
    test('Mobile nav renders correctly', () => {
        const { container } = render(<MobileNav items={items} />);

        expect(container).toMatchSnapshot();
    });

    test('Mobile nav  with submenu shouldrenders correctly', () => {
        const mockedSubmenu = [
            {
                label: 'Blog',
                image: {
                    alt: 'figure',
                    url: 'url_of_image'
                },
                url: '/blog',
                id: 2,
                submenu: [
                    {
                        label: 'PWA Studio',
                        image: {
                            alt: 'figure',
                            url: 'url_of_image'
                        },
                        url: '/pwastudio',
                        id: 3
                    }
                ]
            }
        ];
        const { container } = render(<MobileNav items={mockedSubmenu} />);

        expect(container).toMatchSnapshot();
        expect(container.firstChild.getAttribute('href')).toBeNull();
    });
});
