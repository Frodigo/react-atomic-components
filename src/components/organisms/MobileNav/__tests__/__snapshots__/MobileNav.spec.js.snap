// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile nav should be rendered Mobile nav  with submenu shouldrenders correctly 1`] = `
<div>
  <nav
    class="mobileNav"
  >
    <button
      class="submenuTrigger"
      type="button"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="Blog"
          src="[object Object]"
        />
        <span
          class="label"
        >
          Blog
        </span>
      </span>
    </button>
    <nav
      class="submenu"
    >
      <header
        class="submenuHeader"
      >
        <h2
          class="inverse"
        >
          Read about:
        </h2>
      </header>
      <div
        class="submenuItems"
      >
        <a
          class="submenuItem"
          href="/pwastudio"
        >
          <span
            class="iconContainer"
          >
            <img
              alt="PWA Studio"
              src="[object Object]"
            />
            <span
              class="label"
            >
              PWA Studio
            </span>
          </span>
        </a>
      </div>
      <a
        class="tiertary"
        href="/blog"
        style="margin-top: 30px;"
      >
        Show all
      </a>
    </nav>
  </nav>
</div>
`;

exports[`Mobile nav should be rendered Mobile nav renders correctly 1`] = `
<div>
  <nav
    class="mobileNav"
  >
    <a
      href="/"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="Home"
          src="[object Object]"
        />
        <span
          class="label"
        >
          Home
        </span>
      </span>
    </a>
    <a
      href="/blog"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="Blog"
          src="[object Object]"
        />
        <span
          class="label"
        >
          Blog
        </span>
      </span>
    </a>
    <a
      href="/about-me"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="About"
          src="[object Object]"
        />
        <span
          class="label"
        >
          About
        </span>
      </span>
    </a>
    <a
      href="/get-in-touch"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="Contact"
          src="[object Object]"
        />
        <span
          class="label"
        >
          Contact
        </span>
      </span>
    </a>
    <a
      href="#newsletter"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="Subscribe"
          src="[object Object]"
        />
        <span
          class="label"
        >
          Subscribe
        </span>
      </span>
    </a>
  </nav>
</div>
`;
